src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -442,8 +446,10 @@ def _update_params_for_recommendation_id(
                 self.model_data = compilation_res["ModelArtifacts"]["S3ModelArtifacts"]
                 self.image_uri = compilation_res["InferenceImage"]
 
-        instance_type = recommendation["EndpointConfiguration"]["InstanceType"]
-        initial_instance_count = recommendation["EndpointConfiguration"]["InitialInstanceCount"]
+        instance_type = right_size_recommendation["EndpointConfiguration"]["InstanceType"]
```

Comment:
what if right_size_recommendation is also None? Won't this fail ?

Reply:
This case has been caught by [here](https://github.com/aws/sagemaker-python-sdk/pull/3695/files#diff-7dadd41a1b89ec100e812b257480c34979fc3666e68bef718ad3b0876d71bc53R537).
So `_get_recommendation` will ensure either `right_size_recommendation` or `model_recommendation` exists, nevertheless it will throw exception for `invalid recommendation id`